File: solutions/javascript/general-algos/trees/binaryTreeLevelOrderTraversal.js
const levelOrder = (root) => {
    const result = [];
    
    if (!root) return result;
    
    const queue = new Queue();
    queue.enqueue({
        node: root,
        level: 0
    });
    
    while (!queue.isEmpty()) {
        const { node, level } = queue.dequeue();
        
        const levelInResult = result.length > level;
        if (!levelInResult) result.push([]);
        
        result[level].push(node.val);
        
        const children = [node.left, node.right];
        for (const child of children) {
            if (!child) continue;

            queue.enqueue({
                node: child,
                level: level + 1
            });
        }
    }
    
    return result;
};

File: solutions/python/blind-75/tree/binary_tree_level_order_traversal.py

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        result = []
        
        if not root:
            return result
        
        queue = deque()
        queue.append((root, 0))
        
        while queue:
            node, level = queue.popleft()
            
            level_in_result = len(result) > level
            if not level_in_result:
                result.append([])
            
            result[level].append(node.val)
            
            children = [node.left, node.right]
            for child in children:
                if not child:
                    continue
                
                queue.append((child, level + 1))
        
        return result